### ISSUE TITLE
Intermittent command: Error: Command execution timed out

### ISSUE THREAD

**Issue by shalomb** [aelsabbahy/goss#147]
We intermittently see ``goss`` fail on validating a ``gossfile`` that has quite a few `commands` to run.

Here's an excerpt of an autogenerated gossfile that has the seemingly (??) simple ruby one-liners .. this particular file has ~127 similar commands.

```
--- 
command:

  ? |-
    rvm ruby-1.8.7-p358 do ruby -e '...'
  :
    exit-status: 0
  
  ? |-
    rvm ruby-1.8.7-p358 do ruby -e '...'
  :
   exit-status: 0

  ? |-
    rvm ruby-1.9.2-p320 do ruby -e '...'
  :
    exit-status: 0

  ? |-
    rvm ruby-1.9.2-p320 do ruby -e '...'
  :
    exit-status: 0
  
  ? |-
    rvm ruby-2.1.2 do ruby -e '...'
  :
    exit-status: 0
```

(output (from ansible) trimmed for brevity and to hide sensitive code)
```
$ goss -v
goss version v0.1.5


"bash -c ' gossfile=app.yaml; goss --gossfile \"$gossfile\" validate </dev/null | tee \"/tmp/goss/$gossfile.out\"; 
"............................................................................................................................FFF
Failures:
...
rvm ruby-1.8.7-p358 do ruby -e '....' exit-status: Error: Command execution timed out (10s)
rvm ruby-1.9.2-p320 do ruby -e ' .... ': exit-status: Error: Command execution timed out (10s)
rvm ruby-2.1.2 do ruby -e ' ...': exit-status: Error: Command execution timed out (10s)\n\n
Total Duration: 10.111s\nCount: 127, Failed: 3",
...
```

We'd put the failures down to environmental one-offs (load spikes, etc) but there does seem to be a strong correlation to this happening the more commands the gossfiles have (possibly a GOPROCMAX or similar limitation or a `goss` scheduling limitation). 

1. Examining the load of the VMs around the time of the failures does not reveal anything untowards - but we can't rule that out (see 3).
2. Increasing the timeout (tried upto 60 seconds) does not seem to have an effect - and infact, in the failure path, we have to wait longer for `goss` to fail.
3. `goss` is invoked through `ansible` and played out across many VMs - a failure on one VM almost always coincides with failures on all the other VMs. It would suggest a Hypervisor (Xen here) vCPU scheduling issue and if it indeed is, it's strange that we only see `goss` sensitive to this.
4. Not all commands in the gossfile fail - just a handful between 3 and 10 (out of 127).
5. Failures almost always involve `rvm` :) (and sometimes `curl`) .. it could just be that `rvm` is slower to execute its payload. And repeat failures almost always have the same commands fail.
6. Failures are seen on `gossfiles` with fewer (10s of) commands.

What can we do to try and provide some insights into what's going on? Or should we rethink using this large number of `commands` or `commands` entirely?

**Comment by aelsabbahy** [comment]
Sorry for the inconvenience.

Can you try this experimental release: https://github.com/aelsabbahy/goss/releases/tag/issue%2F147

And use `goss validate --max-concurrent 1` to see if the issue goes away? If it does, you might have to experiment with a max-concurrent that works best for your system. This was hard-coded to 50 before, but this experimental release will allow you to override that.

I'm not 100% certain this is what's causing the issue, but worth a shot.

**Comment by aelsabbahy** [comment]
Did you get a chance to try out the experimental release? I'm probably going to release that feature as part of the next `goss` release since it's a nice feature, but wondering if it's resolved your issue.

**Comment by aelsabbahy** [comment]
Released `--max-concurrent` feature in [v0.2.4](https://github.com/aelsabbahy/goss/releases/tag/v0.2.4). Haven't heard back from you, so closing the issue for now, feel free to re-open if this change has not fixed your issue.
